src/term/check/unbound_vars.rs

```diff
@@ -1,11 +1,13 @@
-use crate::term::{DefinitionBook, LetPat, Name, Term};
+use crate::term::{Book, LetPat, Name, Term};
 use hvmc::run::Val;
 use std::collections::HashMap;
 
-impl DefinitionBook {
+impl Book {
   pub fn check_unbound_vars(&self) -> anyhow::Result<()> {
     for def in self.defs.values() {
-      def.body.check_unbound_vars()?;
+      for rule in def.rules.iter() {
```

Comment:
Either this is called after resolving pattern matching and we know the defs only have 1 rule (and we check that and only access the first rule), or this is called before compiling the pattern matching and then we need to consider the variables in the pattern.
I think the first option is a bit easier